core/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -81,7 +81,8 @@ protected void masterOperation(final ClusterStateRequest request, final ClusterS
         if (request.routingTable()) {
             if (request.indices().length > 0) {
                 RoutingTable.Builder routingTableBuilder = RoutingTable.builder();
-                for (String filteredIndex : request.indices()) {
+                String[] indices = indexNameExpressionResolver.concreteIndexNames(currentState, request);
```

Comment:
Yes. I changed the cluster state api so that indices are resolved for both `metadata` and `routing_table` data.

Before this change, if you have two indices like `index1` and `index2` and you retrieve the cluster state just for metadata and routing table using an index pattern like `index*`, it returns only metadata and corresponding routing tables are missing.

I thought it was coherent to align the behavior.